### PR TITLE
Build enhancements

### PR DESCRIPTION
**Pull request by SergiiDmytruk**
Part of #1002 which can be applied on its own and might be generally useful.

### CODE REVIEW

**Comment by tlaurion** [comment]
Not sure why this has not been picked up by CircleCI, while seeing its building [here](https://app.circleci.com/pipelines/github/SergiiDmytruk/heads?branch=build-enhancements) for all current boards. If no regression will merge.

File: modules/linux
Comment (tlaurion) [inline modules/linux:175]: not -> now

Comment (SergiiDmytruk) [reply modules/linux]: Fixed and force pushed (in case you missed it).

**Comment by macpijan** [comment]
Looks like build has passed?

**Comment by tlaurion** [comment]
@macpijan correct!
@SergiiDmytruk can you launch a rebuild on your CI to have the clear gain on time for comparable (when built from cache)?
@MrChromebox I would merge this PR if there is no objection.

**Comment by MrChromebox** [comment]
LGTM, would want to just boot test here before approving. will do so later today

**Comment by tlaurion** [comment]
Some quick compilation time stats (not so reliable) to compare before after this merge (some delays are added from concurrency limitation applied on credit consumption for CircleCI instances with high commit rate/rebuilds) 

Latest upstream clean build on osresearch CircleCI instance from scratch (no cache used, no concurrency limit): https://app.circleci.com/pipelines/github/osresearch/heads/286/workflows/c31eae0c-834c-41be-a5aa-0ef2fc37c70f
- Build time without concurrency limit: 6h

Latest upstream rebuild (with cache), reusing all prebuilt modules (no concurrency limit): https://app.circleci.com/pipelines/github/osresearch/heads/287/workflows/2aed66e6-a1c2-4750-bf60-0446ec37c4d1
- Build time without concurrency limit: 5h21


----
Fresh build without cache of this PR (with concurrency limit:):https://app.circleci.com/pipelines/github/SergiiDmytruk/heads/39/workflows/bcd1e102-abf6-47a2-b0ff-26cdcfdd8c8d
- build time: 7h22
   -  coreboot-qemu build takes 2h35 which is significantly higher then normal (~1h49)

Rebuild from cache of this PR: https://app.circleci.com/pipelines/github/SergiiDmytruk/heads/39/workflows/afe30d8b-e99d-46f1-a96b-896328952f3a
- build time: 3h47

**Comment by tlaurion** [comment]
@MrChromebox tested https://347-380596387-gh.circle-artifacts.com/0/build/x230-hotp-maximized/heads-x230-hotp-maximized-v0.2.0-1043-g268f628.rom, boots

**Review by MrChromebox — APPROVED** [review]
build/booted tested on a L14, no issues

**Pull request merged**
